http-conduit fails to build with GHC 9.4 and aeson 2.2.0.0

```
[3 of 3] Compiling Network.HTTP.Simple ( Network/HTTP/Simple.hs, /tmp/http-conduit-2.3.8.2/dist-newstyle/build/x86_64-linux/ghc-9.4.5/http-conduit-2.3.8.2/build/Network/HTTP/Simple.o, /tmp/http-conduit-2.3.8.2/dist-newstyle/build/x86_64-linux/ghc-9.4.5/http-conduit-2.3.8.2/build/Network/HTTP/Simple.dyn_o )

Network/HTTP/Simple.hs:106:1: error:
    Could not load module ‘Data.Aeson.Parser’
    It is a member of the hidden package ‘aeson-2.1.2.1’.
    Perhaps you need to add ‘aeson’ to the build-depends in your .cabal file.
    Use -v (or `:set -v` in ghci) to see a list of the files searched for.
    |
106 | import Data.Aeson.Parser (json')
    | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```

The package bounds need to be revised to only allow `aeson <2.2`

--- a/http-conduit/http-conduit.cabal
+++ b/http-conduit/http-conduit.cabal
@@ -42,6 +42,7 @@ library
 
     if flag(aeson)
       build-depends: aeson                 >= 0.8
+                   , attoparsec-aeson      >= 2.1
 
     if !impl(ghc>=7.9)
       build-depends:   void >= 0.5.5
@@ -90,6 +91,7 @@ test-suite test
 
     if flag(aeson)
       build-depends: aeson
+                   , attoparsec-aeson      >= 2.1
 
 source-repository head
   type:     git

--- a/stack.yaml
+++ b/stack.yaml
@@ -1,4 +1,4 @@
-resolver: lts-20.24
+resolver: lts-20.26
 packages:
 - http-client
 - http-client-tls
@@ -18,3 +18,4 @@ extra-deps:
 - warp-tls-3.3.7
 - recv-0.1.0
 - http2-4.1.3
+- attoparsec-aeson-2.1.0.0

